lib/src/type_checker.dart

```diff
@@ -49,18 +49,40 @@ abstract class TypeChecker {
     return results.isEmpty ? null : results.first;
   }
 
-  /// Returns every constant annotating [element] that is this type.
+  /// Returns the first constant annotating [element] that is exactly this type.
+  DartObject firstAnnotationOfExact(Element element) {
+    if (element.metadata.isEmpty) {
+      return null;
+    }
+    final results = annotationsOfExact(element);
```

Comment:
Would be `results.firstWhere((_) => true, orElse: () => null);` IMO what is already in this PR is more readable.